api/v1beta2/awsmachine_types.go

```diff
@@ -197,6 +197,14 @@ type AWSMachineSpec struct {
 	// CapacityReservationID specifies the target Capacity Reservation into which the instance should be launched.
 	// +optional
 	CapacityReservationID *string `json:"capacityReservationId,omitempty"`
+
+	// MarketType specifies the type of market for the EC2 instance. Valid values include:
```

Comment:
Can we describe in gdoc what's the default behaviour if I don't set spot or capacityBlock? i.e on demand